### PR TITLE
Linker assembly toggle

### LINKED ISSUE #121: Include assemblies with sequencing files when linking to sample data (using ngsArchiveLinker)
Imported from GitLab issue #561.  Originally posted on 2017/11/10 10:54AM

### PR DESCRIPTION
**Pull request by tom114**
## Description of changes
Added gui selection for the type of file to export in the ngs linker.  This just helps the user generate the cmd line for them to paste into their terminal.

## Related issue
Fixes #121 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx [hunk 1]
```diff
@@ -27,33 +27,79 @@ const CommandText = styled(Paragraph)`
...
+    </>
+  );
+
   function handleSampleIds(e) {
-    // Post data to get linker command.
+    // Post data to the server to get the linker command.
     const { detail } = e;
     getNGSLinkerCode(detail).then(({ data }) => {
-      Modal.success({
+      scriptString = data;
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx]: You can rename a variable when you are destructuring the object, so:

```
 getNGSLinkerCode(detail).then(({ data : scriptString }) => {
```
would give the same result.

File: src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx [hunk 2]
```diff
@@ -27,33 +27,79 @@ const CommandText = styled(Paragraph)`
...
+  //this will hold the base string of the script with the projects & samples.  The filetypes will be appended before displaying.
+  let scriptString = undefined;
+
+  let modal = undefined; 
+
+  const ModalContents = (
+    { command } 
+  ) => (
+    <>
+      {" "}
```
[the remark this answers is not in the document]
Comment (tom114) [reply src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx]: Removed in 0c2a1d3

File: src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx (hunk 1 above)
Comment (tom114) [reply src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx]: So this wasn't just renaming the variable.  That variable is at https://github.com/phac-nml/irida/pull/638/files#diff-383b10c14b55cb1fcd2260780fbcf323R36 to hold the base of the script string so when it gets updated it doesn't have to repeat the server call.  See how it's building the string to display at https://github.com/phac-nml/irida/pull/638/files#diff-383b10c14b55cb1fcd2260780fbcf323R97.

Let me know if there's a different way to go about this.

**Comment by tom114** [comment]
@joshsadam your PR is merged up into this.  You can re-review if needed.

File: src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx [hunk 3]
```diff
@@ -27,58 +29,85 @@ const CommandText = styled(Paragraph)`
...
+        style={{
+          margin: `0 inherit`,
+          padding: 0,
+          color: grey9
+        }}
+        className="t-linker-btn"
+        onClick={getIds}
+      >
+        {/* constant marginRight here to match old styles */}
+        <CodeOutlined style={{ marginRight: 2 }} />
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx]: Replace this with `IconCode` it's already in the imports, but I must have missed updating it.  Please then remove the import for `CodeOutlined`

Comment (tom114) [reply src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx]: Done in 6cff5b9

**Comment by joshsadam** [comment]
@tom114 sorry is this good to go?  Lost track of this one.

**Comment by tom114** [comment]
It should be done after #634 is done.

**Pull request merged**